src/main/java/org/mule/extensions/jms/internal/operation/profiling/tracing/JmsSpanCustomizer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.internal.operation.profiling.tracing;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.jms.commons.internal.connection.JmsTransactionalConnection;
+import org.mule.sdk.api.runtime.source.DistributedTraceContextManager;
+
+import java.util.Locale;
+
+import javax.jms.ConnectionMetaData;
+import javax.jms.JMSException;
+
+import org.slf4j.Logger;
+
+public abstract class JmsSpanCustomizer {
+
+  private static final Logger LOGGER = getLogger(JmsSpanCustomizer.class);
+  public static final String MESSAGING_SYSTEM = "messaging.system";
+  public static final String MESSAGING_DESTINATION = "messaging.destination";
+
+  protected void customizeSpan(DistributedTraceContextManager distributedTraceContextManager,
+                               JmsTransactionalConnection connection, String destination) {
+    ConnectionMetaData connectionMetaData;
+    try {
+      connectionMetaData = connection.get().getMetaData();
+      distributedTraceContextManager
```

Comment:
verify that the connectionData is not null and add a log for that. Add a general mechanism for only logging problems in the tracing information so that the correct functioning of the connector is not affected by retrieving the data for tracing.

Reply:
I agree with Fabian.